plugin/storage/factory.go

```diff
@@ -118,7 +118,7 @@ func (*Factory) getFactoryOfType(factoryType string) (storage.Factory, error) {
 	case cassandraStorageType:
 		return cassandra.NewFactory(), nil
 	case elasticsearchStorageType, opensearchStorageType:
-		return es.NewFactory(), nil
+		return es.NewFactory(es.PrimaryNamespace), nil
```

Comment:
@yurishkuro can you take another look at the changes? I minimized the code movements to simply remove the `others` field from `Options`. The problem now is that this isn't initializing the CLI flags for `es-archive`. Any thoughts on how to get around this?

Reply:
see https://github.com/jaegertracing/jaeger/pull/6156#discussion_r1827040621

The query service instantiates just a single factory and casts it to ArchiveFactory. With your changes (which are in the right direction, but of insufficient scope) it never gets a chance to create archive CLI flags, because that has to happen via different factories.

Reply:
@yurishkuro is the archive factory only needed for the query service?

Reply:
yes